libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/BotFrameworkPaths.cs

```diff
@@ -9,36 +9,36 @@ public BotFrameworkPaths()
         {
             this.BasePath = "api/";
             this.MessagesPath = "messages";
-            this.ProactiveMessagesPath = "messages/proactive";
+            this.ExternalEventsPath = "events";
```

Comment:
> this.ExternalEventsPath = "events"; [](start = 11, length = 36)

I don't understand this.  we already support external events via posting through directline.  We already have proactive samples on V3 which uses this pattern.  What are we solving here?

Reply:
So let me start by saying there seems to be a lot of confusion around exactly how to trigger proactive conversation. One way is obvious: you're in the midst of another conversation turn. However, what I've seen come up time and again is a question of how to trigger a proactive conversation from some kind of external event. For example, a LogicApp representing some long running operation that just completed and now needs to trigger a proactive conversation with the user to let them know that operation has completed.

This is something I see everyone who's gone slightly more advanced try to do by writing their own controller actions that handle their own proprietary payloads and then they get super confused as to where to get `BotFrameworkAdapter` from to call `ContinueConversation` on (which until PR #553 is in you couldn't even do correctly). So instead of every developer having to go do that themselves, I'm trying to solve that problem "in-the-box" with this integration API which registers a separate handler which expects an `event` activitity to be posted to it as well as some other important information (i.e. `botAppId`) that then establishes the correct contextual information and calls `BotFrameworkAdapter::ProcessActivity` in which the user can be watching for some specific type of `event` and trigger their proactive conversation off of that. Meanwhile, their `IBot` is none-the-wiser as it looks just like any other activity that comes into `OnTurn`.

Now, if you're saying this is something that Direct Line is 100% meant to solve, then that's awesome and I think we just need to find a way to document this scenario better because it seems very common yet nobody has realized that Direct Line is the answer for this. 

I would honestly love to remove this concept from the integration, so if I'm understanding you correctly and Direct Line is what we want to drive people towards for this scenario, then I'll cancel this PR and PR the removal of this extra handler altogether. We would just need to make sure we cover this scenario in the docs space then.... something like "Triggering Proactive Conversations".